src/ctim/schemas/vocabularies.cljc

```diff
@@ -490,3 +490,255 @@
+    "Automated Dynamic Analysis"
+    "Automated Static Analysis"
+    "Automated Static Analysis - Source Code"
+    "Automated Static Analysis - Binary or Bytecode"
+    "Fuzzing"
+    "Manual Analysis"
+    "Manual Dynamic Analysis"
+    "Manual Static Analysis"
+    "Manual Static Analysis - Source Code"
+    "Manual Static Analysis - Binary or Bytecode"
+    "White Box"
+    "Black Box"
+    "Architecture or Design Review"
+    "Dynamic Analysis with Manual Results Interpretation"
+    "Dynamic Analysis with Automated Results Interpretation"
+    "Other"})
+
+(def-enum-type DetectionMethod
+  detection-methods
+  :open? true
+  :description "method used to detect a weakness"
+  :reference "[DetectionMethodEnumeration](https://cwe.mitre.org/documents/schema/#DetectionMethodEnumeration)")
+
+(def detection-effectivness
+  #{"High"
+    "Moderate"
+    "SOAR Partial"
+    "Opportunistic"
+    "Limited"
+    "None"})
+
+(def-enum-type DetectionEffectiveness
+  detection-effectivness
+  :open? true
+  :description (str "level of effectiveness that a detection method may have "
+                    "in detecting an associated weakness")
+  :reference "[DetectionEffectivenessEnumeration](https://cwe.mitre.org/documents/schema/#DetectionEffectivenessEnumeration)")
+
+(def mitigation-strategies
+  #{"Attack Surface Reduction"
+    "Compilation or Build Hardening"
+    "Enforcement by Conversion"
+    "Environment Hardening"
+    "Firewall"
+    "Input Validation"
+    "Language Selection"
+    "Libraries or Frameworks"
+    "Resource Limitation"
+    "Output Encoding"
+    "Parameterization"
+    "Refactoring"
+    "Sandbox or Jail"
+    "Separation of Privilege"})
+
+(def-enum-type MitigationStrategy
+  mitigation-strategies
+  :open? true
+  :description (str "strategy for protecting a system to which a mitigation "
+                    "contributes")
+  :reference "[MitigationStrategyEnumeration](https://cwe.mitre.org/documents/schema/#MitigationStrategyEnumeration)")
+
+(def effectiveness
+  #{"High"
+    "Moderate"
+    "Limited"
+    "Incidental"
+    "Defense in Depth"
+    "None"})
+
+(def-enum-type Effectiveness
+  effectiveness
+  :open? true
+  :description (str "related to how effective a mitigation may be in preventing "
+                    "the weakness")
+  :reference "[EffectivenessEnumeration](https://cwe.mitre.org/documents/schema/#EffectivenessEnumeration)")
+
+(def functional-areas
+  #{"Authentication"
+    "Authorization"
+    "Code Libraries"
+    "Counters"
+    "Cryptography"
+    "Error Handling"
+    "Interprocess Communication"
+    "File Processing"
+    "Logging"
+    "Memory Management"
+    "Networking"
+    "Number Processing"
+    "Program Invocation"
+    "Protection Mechanism"
+    "Session Management"
+    "Signals"
+    "String Processing"
+    "Functional-Area-Independent"})
+
+(def-enum-type FunctionalArea
+  functional-areas
+  :open? true
+  :description "functional area of software appear"
```

Comment:
"Defines the different functional areas of software in which the weakness may appear." is a better description.